crates/storage/db-api/src/table.rs

```diff
@@ -100,6 +100,25 @@ pub trait Table: Send + Sync + Debug + 'static {
     type Value: Value;
 }
 
+/// Trait that provides object-safe access to the table's metadata.
+pub trait TableMetadata: Send + Sync + Debug + 'static {
+    /// The table's name.
+    fn name(&self) -> &'static str;
+
+    /// Whether the table is a `DUPSORT` table.
+    fn is_dupsort(&self) -> bool;
+}
+
+impl<T: Table> TableMetadata for T {
+    fn name(&self) -> &'static str {
+        T::NAME
+    }
+
+    fn is_dupsort(&self) -> bool {
+        T::DUPSORT
+    }
+}
```

Comment:
why is this necessary, isn't this just a subset of `Table`?

Reply:
yes you're right, this is not needed